test/active_model_serializers/test/serializer_test.rb

```diff
@@ -0,0 +1,85 @@
+require 'test_helper'
+
+module ActiveModelSerializers
+  module Test
+    class SerializerTest < ActionController::TestCase
+      include ActiveModelSerializers::Test::Serializer
+
+      class MyController < ActionController::Base
+        TEMPLATE_NAME = 'template'
+        def render_using_serializer
+          render json: Profile.new(name: 'Name 1', description: 'Description 1', comments: 'Comments 1')
+        end
+
+        def render_some_text
+          render text: 'ok'
+        end
+
+        def render_a_template
+          prepend_view_path './test/fixtures'
+          render template: TEMPLATE_NAME
+        end
+      end
+
+      tests MyController
+
+      def test_supports_specifying_serializers_with_a_serializer_class
+        get :render_using_serializer
+        assert_serializer ProfileSerializer
+      end
+
+      def test_supports_specifying_serializers_with_a_regexp
+        get :render_using_serializer
+        assert_serializer(/\AProfile.+\Z/)
+      end
+
+      def test_supports_specifying_serializers_with_a_string
+        get :render_using_serializer
+        assert_serializer 'ProfileSerializer'
+      end
+
+      def test_supports_specifying_serializers_with_a_symbol
+        get :render_using_serializer
+        assert_serializer :profile_serializer
+      end
+
+      def test_supports_specifying_serializers_with_a_nil
+        get :render_some_text
+        assert_serializer nil
+      end
+
+      def test_raises_descriptive_error_message_when_serializer_was_not_rendered
+        get :render_using_serializer
+        e = assert_raise ActiveSupport::TestCase::Assertion do
+          assert_serializer 'PostSerializer'
+        end
+        assert_match 'expecting <"PostSerializer"> but rendering with <["ProfileSerializer"]>', e.message
+      end
+
+      def test_raises_argument_error_when_asserting_with_invalid_object
+        get :render_using_serializer
+        e = assert_raise ArgumentError do
+          assert_serializer Hash
+        end
+        assert_match 'assert_serializer only accepts a String, Symbol, Regexp, ActiveModel::Serializer, or nil', e.message
+      end
+
+      def test_does_not_overwrite_notification_subscriptions
```

Comment:
The intent of this test has changed a bit, and it should probably be renamed.  I'm not really sure why asserting a template was rendered would relate to 'not overwriting notification subscriptions'. The assert template test does use the [`!render_template.action_view'`](https://github.com/rails/rails-controller-testing/blob/f756b33c138c593eabe37f6085f8bac477b99bfe/lib/rails/controller/testing/template_assertions.rb) event and I wonder if it was just that the original implementation used the same setup/teardown names as the rails assertion. (   setup :setup_subscriptions          teardown :teardown_subscriptions )